File: solutions.digamma.damas.jcr/src/main/java/solutions/digamma/damas/jcr/common/ResultPage.java
package solutions.digamma.damas.jcr.common;

import solutions.digamma.damas.entity.Entity;
import solutions.digamma.damas.entity.Page;

import java.util.List;

/**
 * Generic content page.
 *
 * @author Ahmad Shahwan
 */
public class ResultPage<T extends Entity> implements Page<T> {

    private final List<T> content;
    private final int offset;
    private final int total;

    /**
     * Constructor.
     *
     * @param content
     * @param total
     */
    public ResultPage(List<T> content, int offset, int total) {
        this.content = content;
        this.offset = offset;
        this.total = total;
    }

    @Override
    public int getTotal() {
        return this.total;
    }

    @Override
    public int getSize() {
        return this.content.size();
    }

    @Override
    public int getOffset() {
        return this.offset;
    }

    @Override
    public List<T> getObjects() {
        return this.content;
    }
}

File: solutions.digamma.damas.jcr/src/main/kotlin/solutions/digamma/damas/jcr/common/ResultPage.kt
package solutions.digamma.damas.jcr.common

import solutions.digamma.damas.entity.Entity
import solutions.digamma.damas.entity.Page

/**
 * Generic content page.
 *
 * @param content       content of the result page
 * @param total         total size of the result page
 *
 * @author Ahmad Shahwan
 */
internal class ResultPage<T : Entity> (
        private val content: List<T>,
        private val offset: Int, private val total: Int) : Page<T> {

    constructor(content: List<T>) : this(content, 0, content.size)
    constructor(content: List<T>, total: Int) : this(content, total, content.size)

    override fun getTotal(): Int {
        return this.total
    }

    override fun getSize(): Int {
        return this.content.size
    }

    override fun getOffset(): Int {
        return this.offset
    }

    override fun getObjects(): List<T> {
        return this.content
    }
}
